runtime/runtime/src/lib.rs

```diff
@@ -1483,59 +1484,270 @@ impl Runtime {
+    /// insufficient NEAR balance, etc.) will be skipped, producing no receipts or outcomes
+    ///
+    /// In case processing any transaction results in `Err()` the chunk is dropped and
+    /// `RuntimeError` returned.
     fn process_transactions(
         &self,
         processing_state: &mut ApplyProcessingReceiptState,
-        signed_txs: SignedValidPeriodTransactions,
+        tx_vec: SignedValidPeriodTransactions,
         receipt_sink: &mut ReceiptSink,
     ) -> Result<(), RuntimeError> {
-        /// We track the transaction validity in a bit vector of this size. This type informs the
-        /// maximum size of the transactions' chunk processed with each rayon job.
-        type ValidBitmask = u32;
-        /// Avoid the overhead of inter-thread scheduling by processing at least this many
-        /// transactions for each instance of this overhead. This can reduce the number of
-        /// transaction chunks for smaller lists of transactions, however.
-        const MIN_CHUNK_SIZE: usize = 8;
-        /// Avoid splitting transactions into just $NUM_THREADS chunks, as that can result in an
-        /// increased tail latency when one of the threads is slower at processing its chunk
-        /// compared to others (whatever reason may be for that.) Splitting into smaller chunks
-        /// allows the load to be distributed across threads more evenly and any tail latency
-        /// reduced due to the last chunk(s) being smaller.
-        const TARGET_CHUNKS_PER_THREAD: usize = 4;
-        let total = &mut processing_state.total;
         let apply_state = &mut processing_state.apply_state;
-        let state_update = &mut processing_state.state_update;
-        let tx_vec = signed_txs.into_nonexpired_transactions();
+        let state_update = &processing_state.state_update;
         let len = tx_vec.len();
-        let chunk_count_target = rayon::current_num_threads() * TARGET_CHUNKS_PER_THREAD;
-        let chunk_size =
-            (tx_vec.len() / chunk_count_target).clamp(MIN_CHUNK_SIZE, ValidBitmask::BITS as _);
+        const CHUNK_SIZE: usize = 32;
         let protocol_version = processing_state.protocol_version;
         let (valid_masks, (accounts, access_keys)) = rayon::join(
             || {
-                tx_vec
-                    .par_chunks(chunk_size)
-                    .map(|txs| {
-                        let mut valid_mask: ValidBitmask = 0;
-                        for (idx, tx) in txs.iter().enumerate() {
+                let tx_chunks = tx_vec.transactions.par_chunks(CHUNK_SIZE);
+                let validity_chunks =
+                    tx_vec.transaction_validity_check_passed.par_chunks(CHUNK_SIZE);
+                tx_chunks
+                    .zip(validity_chunks)
+                    .map(|(txs, validities)| {
+                        let mut valid_mask: [Option<InvalidTxError>; CHUNK_SIZE] =
```

Comment:
This is 2.5KiB `memmove`d around at least 3 times by my count. And no longer really a "mask" (please rename.) If you absolutely must switch away from a simple bitmask, consider rewriting this iterator in terms of a `flat_map_iter` here rather than flattening later.